sherpa/image/DS9.py

```diff
@@ -333,8 +333,16 @@ def xpaset(cmd, data=None, dataFunc=None, template=_DefTemplate,
         stderr=subprocess.STDOUT,
     )
     try:
-        if isinstance(data, six.string_types):
+        # Python 2 vs 3 requires some complexity here.
+        try:  # Python 2 bytes (which are actually strings)
+            unicode(data, "ascii")  # unicode does not exist in Python 3
             data = bytearray(data, "UTF-8")
+        except:
```

Comment:
Should we only catch the error here rather than catching anything (in case `unicode` in 2.7 raises anything), that is, use

    except NameError:

Similarly, it would be better to just wrap the `unicode` line to minimize problem cases, but perhaps that is too excessive?